### PR TITLE
Port/1.11/hotifxes

### PR DESCRIPTION
**Pull request by Raycoms**
Hotfixes several issues:

- Crashes because of divide/0
- Fixes baker furnaces
- Fixes citizen job assignment
- Fixes home assignment problems

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/colony/CitizenData.java [hunk 1]
```diff
@@ -233,6 +233,70 @@ public void markDirty()
         colony.markCitizensDirty();
     }
 
+    @Override
+    public boolean equals(final Object o)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/colony/CitizenData.java:237]: MAJOR Reduce the number of returns of this method 11, down to the maximum allowed 6. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)

File: src/main/java/com/minecolonies/coremod/colony/Colony.java [hunk 2]
```diff
@@ -57,7 +57,7 @@
 public class Colony implements IColony
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/colony/Colony.java:57]: MAJOR Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 49 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
MAJOR class "Colony" has 92 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)

File: src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingWorker.java [hunk 3]
```diff
@@ -324,7 +323,8 @@ public void onWorldTick(@NotNull final TickEvent.WorldTickEvent event)
 
         // If we have no active worker, grab one from the Colony
         // TODO Maybe the Colony should assign jobs out, instead?
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingWorker.java:325]: INFO Complete the task associated to this TODO comment. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1135)

File: src/main/java/com/minecolonies/coremod/colony/buildings/BuildingHome.java [hunk 4]
```diff
@@ -164,6 +164,7 @@ public void removeCitizen(@NotNull final CitizenData citizen)
         {
             citizen.setHomeBuilding(null);
             residents.remove(citizen);
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/colony/buildings/BuildingHome.java:166]: MINOR This call to "remove()" may be a performance hot spot if the collection is large. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2250)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 95 issues
* ![MAJOR][MAJOR] 49 major
* ![MINOR][MINOR] 43 minor
* ![INFO][INFO] 3 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [CitizenData.java#L26](https://github.com/Minecolonies/minecolonies/blob/70826272a1707a0a11a4416c14c742af96475e4f/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L26): class "CitizenData" has 45 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [CitizenData.java#L518](https://github.com/Minecolonies/minecolonies/blob/70826272a1707a0a11a4416c14c742af96475e4f/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L518): Use the "equals" method if value comparison was intended. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1698)
1. ![MAJOR][MAJOR] [CitizenData.java#L523](https://github.com/Minecolonies/minecolonies/blob/70826272a1707a0a11a4416c14c742af96475e4f/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L523): Use the "equals" method if value comparison was intended. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1698)
1. ![MAJOR][MAJOR] [CitizenData.java#L578](https://github.com/Minecolonies/minecolonies/blob/70826272a1707a0a11a4416c14c742af96475e4f/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L578): Use the "equals" method if value comparison was intended. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1698)
1. ![MAJOR][MAJOR] [CitizenData.java#L582](https://github.com/Minecolonies/minecolonies/blob/70826272a1707a0a11a4416c14c742af96475e4f/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L582): Use the "equals" method if value comparison was intended. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1698)
1. ![MAJOR][MAJOR] [Colony.java](https://github.com/Minecolonies/minecolonies/blob/70826272a1707a0a11a4416c14c742af96475e4f/src/main/java/com/minecolonies/coremod/colony/Colony.java): This file has 1.372 lines, which is greater than 1.000 authorized. Split it into smaller files. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [Colony.java#L226](https://github.com/Minecolonies/minecolonies/blob/70826272a1707a0a11a4416c14c742af96475e4f/src/main/java/com/minecolonies/coremod/colony/Colony.java#L226): Assign this magic number 5 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L626](https://github.com/Minecolonies/minecolonies/blob/70826272a1707a0a11a4416c14c742af96475e4f/src/main/java/com/minecolonies/coremod/colony/Colony.java#L626): Reduce the number of returns of this method 11, down to the maximum allowed 6. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)
1. ![MAJOR][MAJOR] [Colony.java#L1075](https://github.com/Minecolonies/minecolonies/blob/70826272a1707a0a11a4416c14c742af96475e4f/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1075): The Cyclomatic Complexity of this method "onWorldTick" is 18 which is greater than 15 authorized. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![MAJOR][MAJOR] [Colony.java#L1264](https://github.com/Minecolonies/minecolonies/blob/70826272a1707a0a11a4416c14c742af96475e4f/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1264): Assign this magic number 16 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Review by marchermans — APPROVED** [review]

**Pull request merged**